applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaFixedConstraint.inl

```diff
@@ -342,12 +346,12 @@ void FixedConstraintInternalData<gpu::cuda::CudaVec1fTypes>::projectResponse(Mai
 
 template <>
 void FixedConstraintInternalData<gpu::cuda::CudaVec3fTypes>::projectResponse(Main* m, VecDeriv& dx)
-{
+{    
     Data& data = *m->data;
     if (m->d_fixAll.getValue())
-        FixedConstraintCuda3f_projectResponseContiguous(dx.size(), ((float*)dx.deviceWrite()));
-    else if (data.minIndex >= 0)
-        FixedConstraintCuda3f_projectResponseContiguous(data.maxIndex-data.minIndex+1, ((float*)dx.deviceWrite())+3*data.minIndex);
+        FixedConstraintCuda3f_projectResponseContiguous(dx.size(), ((float*)dx.deviceWrite()));        
+    else if (data.minIndex != sofa::InvalidID)
```

Comment:
@hugtalbot here is the bug I fixed but actually the code was certainly failing at other places because before data.minIndex was an int. So the test >= 0 means a valid int value. 
But as data.minIndex was changed to a Sofa:Index (aka unsigned int) in a previous PR and still initialised to -1, the invalid value was 42x10^9 something (max unsigned int). So the test >=0 was always true. 
So to fix I changed all the if condition to test against an InvalidID value.